MapboxCoreNavigation/RouteController.swift

```diff
@@ -527,6 +533,13 @@ extension RouteController: CLLocationManagerDelegate {
         
         let filteredLocations = locations.filter { $0.isQualified }
         
+        // Always update no matter if location is bad
+        if let lastLocation = filteredLocations.last {
+            sessionState.pastLocations.push(lastLocation)
+        } else if let lastLocation = locations.last {
```

Comment:
a) Should we exit early if there is no final location?
b) Should we just add all the locations in the array?

Reply:
Yeah let's do b.